framework/reflect.go

```diff
@@ -73,6 +77,19 @@ func (m *Import) reflectValue(v reflect.Value) (reflect.Value, error) {
 }
 
 func (m *Import) reflectMap(mv reflect.Value) (reflect.Value, error) {
+	// Create a new map for this. This avoids conflicts and panics on shared
+	// data, and ensures we aren't altering data in the original namespace.
+	// map[string]interface{} is always used, regardless of the actual type of
+	// the map sent along. This prevents type panics.
+	//
+	// Do a quick check to see if we have a zero-value map (nil map). If we do,
+	// return that.
+	if mv == reflect.Zero(mv.Type()) {
```

Comment:
If we end up here, are we not doubling up on calls to creating a map? Could we maybe do.

```suggestion
	if zm := reflect.Zero(mv.Type()); mv == zm {
		return zm, nil
```

Reply:
The intention was to create a new map regardless for correctness, but I don't see an issue with this either.

There is also an `IsZero()` that can be used, so I'll use that... probably should have used it in the first place anyway.